### ISSUE TITLE
Filter with "Dynamically search all filter values" enabled selects all values if you select only one specific

### ISSUE THREAD

**Issue by kapustor** [apache/superset#23956]
There is a dashboard with filters. There is a filter with "Dynamically search all filter values" - it is important to enable this feature for the filter.  If you type an exact value in the filter box (when only one value is suggested), and then select the suggested value, all values will be selected instead.

This error didn't exist in Superset 1.5.3 but exists in 2.1.0.

#### How to reproduce the bug

1. Go to SQL Lab
2. Type a query:
```
select ('Chip') as name
union all
select ('Dale') as name
```
3. Save as a dataset
4. Create an empty dashboard
5. Add a filter on column 'name' to the dashboard
6. Enable "Dynamically search all filter values" for the filter and save fiters
7. Now type "Chip" in the filter box - its is important to enter enough characters to only one suggestion to be made. If more the one suggestions are shown, the bug doesn't reproduce.
8. In a moment all values will be selected instead of "Chip", which is not correct.

### Expected results

Only one value should be selected

### Actual results

All values are selected instead

#### Screenshots

![image](https://user-images.githubusercontent.com/25294423/236647910-2c163ad8-5b75-4c22-a3f9-07e6a1518325.png)
![image](https://user-images.githubusercontent.com/25294423/236647920-71bddf7f-8d19-4625-93fa-1610775b39f5.png)
![image](https://user-images.githubusercontent.com/25294423/236647929-133053dd-2cc4-466b-b202-9737b242300b.png)
![image](https://user-images.githubusercontent.com/25294423/236647940-0aff0357-a713-4f7b-86d4-5ab296d29bb7.png)

### Environment

- browser type and version: `Chrome 112.0.5615.138`
- superset version: `2.1.0` (doesn't reproduce in 1.5.3)
- python version: `3.8`
- node.js version: `N/A`
- any feature flags active: None

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [X] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by PedroMartinSteenstrup** [comment]
We encountered the same.
When the `Dynamically search all filter values` is ticked in the filter, input a unique value results in a thousand being selected.

When activated, it makes a check on the cache
```
{
  "datasource": {
    "id": 1997,
    "type": "table"
  },
  "force": false,
  "queries": [
    {
      "filters": [
        {
          "col": "OFFER_ID",
          "op": "ILIKE",
          "val": "%1aad4ce5a3168f49%"
        }
      ],
      "extras": {
        "having": "",
        "where": ""
      },
      "applied_time_extras": {},
      "columns": [
        "OFFER_ID"
      ],
      "metrics": [],
      "orderby": [
        [
          "OFFER_ID",
          true
        ]
      ],
      "annotation_layers": [],
      "row_limit": 1000,
      "series_limit": 0,
      "order_desc": true,
      "url_params": {
        "native_filters_key": "hZfvKcnkPHcIk8mx_aRM8m-CyMqIJ47t0AxfXiy-mfKuhBNmmL3WEc0wD8-ufqmX"
      },
      "custom_params": {},
      "custom_form_data": {}
    }
  ],
  "form_data": {
    "enableEmptyFilter": false,
    "defaultToFirstItem": false,
    "multiSelect": true,
    "searchAllOptions": true,
    "inverseSelection": false,
    "datasource": "1997__table",
    "groupby": [
      "OFFER_ID"
    ],
    "adhoc_filters": [],
    "extra_filters": [],
    "extra_form_data": {},
    "metrics": [
      "count"
    ],
    "row_limit": 1000,
    "showSearch": true,
    "url_params": {
      "native_filters_key": "hZfvKcnkPHcIk8mx_aRM8m-CyMqIJ47t0AxfXiy-mfKuhBNmmL3WEc0wD8-ufqmX"
    },
    "inView": true,
    "viz_type": "filter_select",
    "type": "NATIVE_FILTER",
    "force": false,
    "result_format": "json",
    "result_type": "full"
  },
  "result_format": "json",
  "result_type": "full"
}
```

Response:
```
{"result": [{"cache_key": "4e97d83bceb40003f1bd3becb38d00ef", "cached_dttm": "2023-05-30T14:30:54", "cache_timeout": 86400, "applied_template_filters": [], "annotation_data": {}, "error": null, "is_cached": true, "query": "SELECT \"OFFER_ID\" AS \"OFFER_ID\"\nFROM\n  (SELECT *\n   FROM reporting_daily) AS virtual_table\nWHERE lower(\"OFFER_ID\") LIKE lower('%1aad4ce5a3168f49%')\nGROUP BY \"OFFER_ID\"\nORDER BY \"OFFER_ID\" ASC\nLIMIT 1000", "status": "success", "stacktrace": null, "rowcount": 1, "from_dttm": null, "to_dttm": 1685404800000.0, "label_map": {"OFFER_ID": ["OFFER_ID"]}, "colnames": ["OFFER_ID"], "indexnames": [0], "coltypes": [1], "data": [{"OFFER_ID": "1aad4ce5a3168f49"}], "result_format": "json", "applied_filters": [{"column": "OFFER_ID"}], "rejected_filters": []}]}
```

## Environment
- Google Chrome - Version 113.0.5672.126 (Official Build) (arm64)
- Superset Version: 2.1.0 via Helm
- Feature flags:
```
  FEATURE_FLAGS = {
          "ALERT_REPORTS": True,
          "ALLOW_ADHOC_SUBQUERY": True,
          "ENABLE_TEMPLATE_PROCESSING": True,
          "DASHBOARD_NATIVE_FILTERS": True,
          "DASHBOARD_CROSS_FILTERS": True,
          "DASHBOARD_CACHE": True,
          "GENERIC_CHART_AXES": True,
          "SQLLAB_BACKEND_PERSISTENCE": False
      }
```

**Comment by ASchmidtGit** [comment]
Is there a workaround? E.g. disabling the "select all" option?

**Comment by kapustor** [comment]
Was fixed for me in 2.1.1.

**Comment by rusackas** [comment]
This is likely fixed by now, and is pretty out of date if not. If people are still encountering this in current versions (3.x) please open a new Issue or a PR to address the problem. ... or I guess we can re-open this if nothing has changed. That last comment has me optimistic, so I'll close this :)
